rust-sdk/whirlpool/src/tests/token_test.rs

```diff
@@ -0,0 +1,126 @@
+use super::utils::{rpc::RpcContext, token::*, token_extensions::*};
+use crate::NativeMintWrappingStrategy;
+use serial_test::serial;
+use solana_sdk::signer::Signer;
+use spl_token::native_mint;
+use spl_token_2022::{extension::ExtensionType, ID as TOKEN_2022_PROGRAM_ID};
+
+#[tokio::test]
+#[serial]
+async fn test_no_tokens() {
+    let ctx = RpcContext::new().await;
+    let result = prepare_token_accounts_instructions(&ctx, ctx.signer.pubkey(), vec![])
+        .await
+        .unwrap();
+
+    assert_eq!(result.create_instructions.len(), 0);
+    assert_eq!(result.cleanup_instructions.len(), 0);
+    assert_eq!(result.token_account_addresses.len(), 0);
+}
+
+#[tokio::test]
+#[serial]
+async fn test_native_mint_wrapping_none() {
+    let ctx = RpcContext::new().await;
+    crate::set_native_mint_wrapping_strategy(NativeMintWrappingStrategy::None).unwrap();
+
+    let result = prepare_token_accounts_instructions(
+        &ctx,
+        ctx.signer.pubkey(),
+        vec![TokenAccountStrategy::WithoutBalance(native_mint::ID)],
+    )
+    .await
+    .unwrap();
+
+    assert_eq!(result.create_instructions.len(), 1); // Create ATA
```

Comment:
This currently only asserts the length of the instructions (but the instructions might be wrong). Instead we can actually send the instructions to BanksClient (like we do in the e2e tests) we can assert the pre- and post- state of BanksClient. That should give us more confidence that the instructions created by the prepare_token_accunts_instructions are actually correct/valid

Reply:
Ok added more sophisticated asserts